include/dt-bindings/pinctrl/stm32-pinctrlf1.h

```diff
@@ -75,10 +75,35 @@
  * registers for particular pin.
  */
 
-#define STM32_PIN_USART_TX	(STM32_MODE_OUTPUT | STM32_CNF_ALT_FUNC | STM32_CNF_PUSH_PULL)
-#define STM32_PIN_USART_RX	(STM32_MODE_INPUT | STM32_CNF_IN_FLOAT)
-#define STM32_PIN_I2C		(STM32_MODE_OUTPUT | STM32_CNF_ALT_FUNC | STM32_CNF_OPEN_DRAIN)
-#define STM32_PIN_PWM		(STM32_MODE_OUTPUT | STM32_CNF_ALT_FUNC | STM32_CNF_PUSH_PULL)
+#define STM32_PIN_USART_TX		(STM32_MODE_OUTPUT | STM32_CNF_ALT_FUNC | STM32_CNF_PUSH_PULL)
+#define STM32_PIN_USART_RX		(STM32_MODE_INPUT | STM32_CNF_IN_FLOAT)
+#define STM32_PIN_I2C			(STM32_MODE_OUTPUT | STM32_CNF_ALT_FUNC | STM32_CNF_OPEN_DRAIN)
+#define STM32_PIN_PWM			(STM32_MODE_OUTPUT | STM32_CNF_ALT_FUNC | STM32_CNF_PUSH_PULL)
+#define STM32_PIN_SPI_MASTER_SCK	(STM32_MODE_OUTPUT | STM32_CNF_ALT_FUNC | STM32_CNF_PUSH_PULL)
+#define STM32_PIN_SPI_SLAVE_SCK		(STM32_MODE_INPUT | STM32_CNF_IN_FLOAT)
+#define STM32_PIN_SPI_MASTER_MOSI	(STM32_MODE_OUTPUT | STM32_CNF_ALT_FUNC | STM32_CNF_PUSH_PULL)
+#define STM32_PIN_SPI_SLAVE_MOSI	(STM32_MODE_INPUT | STM32_CNF_IN_FLOAT)
+#define STM32_PIN_SPI_MASTER_MISO	(STM32_MODE_INPUT | STM32_CNF_IN_FLOAT)
+#define STM32_PIN_SPI_SLAVE_MISO	(STM32_MODE_OUTPUT | STM32_CNF_ALT_FUNC | STM32_CNF_PUSH_PULL)
 
+/*
+ * Reference manual (RM0008)
+ * Section 25.3.1: Slave select (NSS) pin management
+ *
+ * Hardware NSS management:
+ * - NSS output disabled: allows multimaster capability for devices operating
+ *   in master mode.
+ * - NSS output enabled: used only when the device operates in master mode.
+ *
+ * Software NSS management:
+ * - External NSS pin remains free for other application uses.
+ *
+ */
+
+/* Hardware master NSS output disabled */
+#define STM32_PIN_SPI_MASTER_NSS	(STM32_MODE_INPUT | STM32_CNF_IN_FLOAT)
+/* Hardware master NSS output enabled */
+#define STM32_PIN_SPI_MASTER_NSS_OE	(STM32_MODE_OUTPUT | STM32_CNF_ALT_FUNC | STM32_CNF_PUSH_PULL)
```

Comment:
@erwango Is this clear enough? STM32_PIN_SPI_MASTER_NSS_OUTPUT_ENABLED is too long.

Reply:
Yes this is fine, thanks